src/basic/terminal-util.c

```diff
@@ -306,7 +306,28 @@ int reset_terminal_fd(int fd, bool switch_to_text) {
         termios.c_cc[VMIN]   = 1;
 
         r = RET_NERRNO(tcsetattr(fd, TCSANOW, &termios));
+        if (r < 0) {
+                log_debug_errno(r, "Failed to set terminal parameters: %m");
+                goto finish;
+        }
+
+        if (!terminal_is_dumb()) {
+                r = fd_nonblock(fd, true);
+                if (r < 0) {
+                        log_debug_errno(r, "Failed to set terminal to non-blocking mode: %m");
+                        goto finish;
+                }
+
+                 /* Enable line wrapping. */
+                (void) loop_write_full(fd, "\033[?7h", SIZE_MAX, 50 * USEC_PER_MSEC);
 
+                if (r > 0)
+                        r = fd_nonblock(fd, false);
+                if (r < 0) {
+                        log_debug_errno(r, "Failed to set terminal back to blocking mode: %m");
+                        goto finish;
+                }
```

Comment:
i think this should be indented one level, i.e. be part of the `if (r > 0)` block, given that this is error handling of the fd_nonblock(false) here, not of the earlier fd_nonblock(true)